knowledge-asset-tools.js

```diff
@@ -130,7 +136,166 @@ async function peerId2Hash(peerId) {
   return ethers.utils.sha256(ethers.utils.toUtf8Bytes(peerId));
 }
 
+function groupNquadsBySubject(nquadsArray) {
+  const store = new N3.Store();
+  const parser = new N3.Parser({ format: "star" });
+
+  nquadsArray.forEach((quad) => {
+    try {
+      const parsedQuad = parser.parse(quad);
+      parsedQuad.forEach((quad) => store.addQuad(quad));
+    } catch (error) {
+      console.error("Error parsing quad:", quad);
+      console.error(error);
+    }
+  });
+
+  function groupTriples(triples) {
```

Comment:
lets have this function somewhere else instead of inside the other function, it could be useful